React/React-Core-umbrella.h file not found - Integrating Expo tools into existing native apps

### Summary

While following the instructions on [Integrating Expo tools into existing native apps](https://docs.expo.dev/brownfield/overview/) I ran into the following compilation error

`../Pods/Headers/Public/ExpoModulesCore/ExpoModulesCore/ExpoBridgeModule.h:3:9: error: 'React/React-Core-umbrella.h' file not found
#import <React/React-Core-umbrella.h>
        ^
<unknown>:0: error: could not build Objective-C module 'ExpoModulesCore'`

I first encountered the issue on my existing Swift/Obj-c project and then reproduced on a new sample project

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

iOS

### Package versions

_No response_

### Environment

```text
expo-env-info 1.2.2 environment info:
    System:
      OS: macOS 15.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 23.7.0 - ~/.asdf/installs/nodejs/23.7.0/bin/node
      Yarn: 1.22.22 - /opt/homebrew/bin/yarn
      npm: 10.9.2 - ~/.asdf/plugins/nodejs/shims/npm
    Managers:
      CocoaPods: 1.16.2 - /Users/j.n/.asdf/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.2, iOS 18.2, macOS 15.2, tvOS 18.2, visionOS 2.2, watchOS 11.2
    IDEs:
      Android Studio: 2024.2 AI-242.23726.103.2422.12816248
      Xcode: 16.2/16C5032a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^52.0.32 => 52.0.32 
      react: 18.3.1 => 18.3.1 
      react-native: 0.77.0 => 0.77.0 
    Expo Workflow: managed
```

### Reproducible demo

https://github.com/JamieScanlon/expo-bug-report-s1f10

### Stacktrace (if a crash is involved)

```text

```

--- a/docs/pages/brownfield/installing-expo-modules.mdx
+++ b/docs/pages/brownfield/installing-expo-modules.mdx
@@ -166,24 +166,34 @@ Override `onConfigurationChanged` if you have not done so already.
 Add the following to your `Podfile` in the **ios** directory:
 
 <DiffBlock
-    raw={`
+  raw={`
 diff --git a/ios/Podfile b/ios/Podfile
-index 2d3a979..825a9d4 100644
+index f991b7b..17c24b0 100644
 --- a/ios/Podfile
 +++ b/ios/Podfile
-@@ -1,3 +1,4 @@
+@@ -1 +1,4 @@
++# Expo requires
 +require File.join(File.dirname(\`node --print "require.resolve('expo/package.json')"\`), "scripts/autolinking")
- require Pod::Executable.execute_command('node', ['-p',
-   'require.resolve(
-   "react-native/scripts/react_native_pods.rb",
-@@ -14,6 +15,7 @@ if linkage != nil
- end
-
-target '<YourAppTarget>' do
-
-+use_expo_modules!
-config = use_native_modules!
-`}
++
+ # Resolve react_native_pods.rb with node to allow for hoisting
+@@ -17,3 +20,16 @@ end
+ target 'SimpleTest' do
+-  config = use_native_modules!()
++  # Need to be added inside the target block
++  use_expo_modules!
++
++  config_command = [
++    'node',
++    '--no-warnings',
++    '--eval',
++    'require(require.resolve(\'expo-modules-autolinking\', { paths: [require.resolve(\'expo/package.json\')] }))(process.argv.slice(1))',
++    'react-native-config',
++    '--json',
++    '--platform',
++    'ios'
++  ]
++  config = use_native_modules!(config_command)
+ `}
 />
 
 </Step>

--- a/docs/pages/brownfield/installing-expo-modules.mdx
+++ b/docs/pages/brownfield/installing-expo-modules.mdx
@@ -177,7 +177,7 @@ index f991b7b..17c24b0 100644
 +
  # Resolve react_native_pods.rb with node to allow for hoisting
 @@ -17,3 +20,16 @@ end
- target 'SimpleTest' do
+ target '<YourAppTarget>' do
 -  config = use_native_modules!()
 +  # Need to be added inside the target block
 +  use_expo_modules!

